R/susie_new.R

```diff
@@ -0,0 +1,147 @@
+susie_new = function (X, y, L = min(10,ncol(X)),
+                  scaled_prior_variance = 0.2,
+                  residual_variance = NULL,
+                  prior_weights = NULL,
+                  null_weight = 0,
+                  standardize = TRUE,
+                  intercept = TRUE,
+                  estimate_residual_variance = TRUE,
+                  estimate_prior_variance = TRUE,
+                  estimate_prior_method = c("optim", "EM", "simple"),
+                  check_null_threshold = 0,
+                  prior_tol = 1e-9,
+                  residual_variance_upperbound = Inf,
+                  model_init = NULL,
+                  coverage = 0.95,
+                  min_abs_corr = 0.5,
+                  #median_abs_corr = NULL,
+                  compute_univariate_zscore = FALSE,
+                  na.rm = FALSE,
+                  max_iter = 100,
+                  tol = 1e-3,
+                  verbose = FALSE,
+                  track_fit = FALSE,
+                  residual_variance_lowerbound = var(drop(y))/1e4,
+                  refine = FALSE,
+                  n_purity = 100) {
+
+  # Estimate Prior Variance Method
+  estimate_prior_method <- match.arg(estimate_prior_method)
+
+  # Handle null weight
+  if (is.numeric(null_weight) && null_weight == 0)
+    null_weight <- NULL
+
+  if (!is.null(null_weight)) {
+    if (!is.numeric(null_weight) || null_weight <= 0 || null_weight >= 1)
+      stop("null_weight must be between 0 and 1 (exclusive)")
+
+    if (is.null(prior_weights))
```

Comment:
these kind of checks can go into constructor instead? You check inputs there anyways